Dropdown list (Mobile): Min width

grzegorz-czajkowski opened the pull request:
Two commits:
 * UIComponents (Mobile): add Dropdown list (Min width)
 * Dropdown list (Mobile): apply alignment according to guide

grzegorz-czajkowski committed:
UIComponents (Mobile): add Dropdown list (Min width)

[Issue] https://github.com/Samsung/TAU/issues/771
[Problem] Missing Dropdown list (Min width) example
[Solution] Add Dropdown list (Min width) example

Signed-off-by: Grzegorz Czajkowski <g.czajkowski@samsung.com>

grzegorz-czajkowski committed:
Dropdown list (Mobile): apply alignment according to guide

[Issue] https://github.com/Samsung/TAU/issues/771
[Problem] Dropdown list is showed at element position to
	wich select element is attached. According to guide,
	dropdown list should be showed on most left or most
	right position (see appbar guide).
	Top and buttom margins are always added whereas they should
	not appear on List elements.
[Solution] Show dropdown menu on left or right side.
	Do not add top and bottom margins if dropdown menu is showed
	on list.

Signed-off-by: Grzegorz Czajkowski <g.czajkowski@samsung.com>

DohyungLim commented on src/js/profile/mobile/widget/DropdownMenu.js:
```diff
@@ -1010,17 +1011,19 @@
+				offsetLeft = self._horizontalDirection === "right" ? window.screen.width - width : 0;
 				optionStyle = "left: " + offsetLeft + "px; ";
+
 				if (options.inline === true) {
```
Type is boolean, do I need "true"? If it can be omitted, nothing is concise.

grzegorz-czajkowski replied:
Are you refering to this line?
`if (options.inline === true)`
I am not changing this and could be refactored separately.

TomaszLukawskiSam replied:
I agree with Grzegorz. This is out of scope this PR.

DohyungLim reviewed (commented):
1 comment need to check.

grzegorz-czajkowski commented:
Solved most issues related to alignment position
 - added top and bottom margins if dropdown is only opened from non list item element (for example appabar)
 - dropdow list is shown of left or right most position based on click position
 - fixed alignment on list (avoid addition 12px from top, added from js code).
Created additional issue https://github.com/Samsung/TAU/issues/808 where we need to take into consideration list position to always show dropdown menu within list.

TomaszLukawskiSam commented on src/css/profile/mobile/common/dropdownmenu.less:
```diff
@@ -194,6 +191,11 @@
 		border-top: 1 * @px_base solid @color_dropdownmenu_list_group_stroke;
 		border-bottom: 1 * @px_base solid @color_dropdownmenu_list_group_stroke;
 	}
+	// top, bottom classes just describes direction in which menu is shown.
```
you don't need define empty styles.

grzegorz-czajkowski replied:
Right but they are still needed.
Those classes (top or bottom) are applied by DropdownMenu.js and are used by css to animate showing menu in specified direction, for example:
```
        &.ui-dropdownmenu-options-opening.ui-dropdownmenu-options-top .ui-dropdownmenu-options {
                .animation(open-to-top 300ms);
        }
```

TomaszLukawskiSam commented on src/js/profile/mobile/widget/DropdownMenu.js:
```diff
@@ -578,30 +587,28 @@
-			 * Return offset of element
-			 * @method getOffsetOfElement
+			 * Return top offset of element
+			 * @method getTopOffsetOfElement
```
You modified description but method name remains old.

grzegorz-czajkowski replied:
Done.

TomaszLukawskiSam commented on src/js/profile/mobile/widget/DropdownMenu.js:
```diff
@@ -1073,12 +1086,14 @@
 			 * @protected
```
@param {event} [clickEvent=null] click event

grzegorz-czajkowski replied:
Done.

TomaszLukawskiSam reviewed (approved):
Works good.
